core/src/test/java/com/google/googlejavaformat/java/FormatterTest.java

```diff
@@ -418,4 +418,78 @@ public void throwsFormatterException() throws Exception {
     } catch (FormatterException expected) {
     }
   }
+
+  @Test
+  public void indentsIfStatementsWithoutBracesButWithCommentsOnTheSameLine()
+      throws FormatterException {
+    assertThat(
+            new Formatter()
+                .formatSource(
+                    "public class CommentedTest {\n"
+                        + "\n"
+                        + "  public boolean compare(ProglangType other) {\n"
+                        + "    if (this == other) // comment\n"
+                        + "    return true;\n"
+                        + "    return false;\n"
+                        + "  }\n"
+                        + "}\n"))
+        .isEqualTo(
+            "public class CommentedTest {\n"
+                + "\n"
+                + "  public boolean compare(ProglangType other) {\n"
+                + "    if (this == other) // comment\n"
+                + "     return true;\n"
```

Comment:
I think that the line generated here by google-java-format should be indented +1.

To give a visual example, the program currently outputs the following,

```java
if (this == other) // comment
 return true;
```

but AFAICT it should output the following instead.

```java
if (this == other) // comment
  return true;
```

Reply:
You're right, it's not using the `plusTwo` indent that I've set, but instead it uses the `" "` flat. I'm looking into this. I think I've found where to fix it. Thank you!

Reply:
@kitonjicsm You're very welcome! :)